tests/unit/conditions/test_evm_condition_serializers.py

```diff
@@ -22,14 +22,24 @@
 from nucypher.policy.conditions.lingo import ConditionLingo
 
 
+def test_evm_condition_function_abi(TStaking_data):
+    original_data = TStaking_data
+    condition = ContractCondition.from_json(original_data)
+    serialized_data = condition.to_json()
+
+    deserialized_data = json.loads(serialized_data)
+    assert deserialized_data["functionAbi"] == condition.function_abi
+
+
 def test_evm_condition_json_serializers(ERC1155_balance_condition_data):
     original_data = ERC1155_balance_condition_data
     condition = ContractCondition.from_json(original_data)
     serialized_data = condition.to_json()
 
-    # TODO functionAbi is present in serialized data
     deserialized_data = json.loads(serialized_data)
     deserialized_data.pop("functionAbi")
+    deserialized_data["returnValueTest"].pop("key")
```

Comment:
Why is this needed?

The original reason for L40, is that the ContractCondition automatically derives the functionABI from the standard contract type. However `key` isn't required and isn't specified in the original condition's return value test, so why is it part of the serialized data?

Reply:
I think if `key` is optional, and doesn't have a value (or is `None`), it shouldn't be serialized.

We should also fix the serialization of `functionAbi` while we are at it. That was a stop gap for the demo. If the standard contract type and method are provided, there is no need to serialize the `functionAbi` property either.

@KPrasch wdyt?

Reply:
Marshmallow schemas can be configured to exclude fields that have a `None` value during de/serialization.  This is currently implemented in `ContractCondition`:

```python
class ContractCondition(RPCCondition):
    class Schema(RPCCondition.Schema):
        SKIP_VALUES = (None,)
```

The same can be done for `key` in `ReturnValueTest.ReturnValueTestSchema` which will avoid the situation seen in this test with `.pop`:

```python

class ReturnValueTestSchema(CamelCaseSchema):
    SKIP_VALUES = (None,)
    ...
    key = fields.Raw(allow_none=True)
```